In [1]:
# Importing packages
import pandas as pd

# Assessing Data

In [4]:
# Reading the csv file
df = pd.read_csv('3051011_PortoAlegre_Diaria.csv', decimal = ',')

In [5]:
df.head()

Data  Valor  Mês   Ano Mês-Ano  ano_valido  Valor_ano_valido  \
0      01/01/61    0.4    1  1961  jan-61      1961.0               0.4   
1      02/01/61    0.0    1  1961  jan-61      1961.0               0.0   
2      03/01/61    0.2    1  1961  jan-61      1961.0               0.2   
3      04/01/61    2.4    1  1961  jan-61      1961.0               2.4   
4      05/01/61    0.0    1  1961  jan-61      1961.0               0.0   
...         ...    ...  ...   ...     ...         ...               ...   
20600  27/05/17   51.4    5  2017  mai-17         NaN               NaN   
20601  28/05/17   15.6    5  2017  mai-17         NaN               NaN   
20602  29/05/17    2.7    5  2017  mai-17         NaN               NaN   
20603  30/05/17    0.4    5  2017  mai-17         NaN               NaN   
20604  31/05/17    3.7    5  2017  mai-17         NaN               NaN   

       Verific_7dias  Média 7dias  Unnamed: 9  Unnamed: 10  Unnamed: 11  \
0                  1     1.200000         NaN          NaN          NaN   
1                  1     1.142857         NaN          NaN          NaN   
2                  1     1.142857         NaN          NaN          NaN   
3                  1     4.957143         NaN          NaN          NaN   
4                  1     6.000000         NaN          NaN          NaN   
...              ...          ...         ...          ...          ...   
20600              0          NaN         NaN          NaN          NaN   
20601              0          NaN         NaN          NaN          NaN   
20602              0          NaN         NaN          NaN          NaN   
20603              0          NaN         NaN          NaN          NaN   
20604              0          NaN         NaN          NaN          NaN   

       Unnamed: 12  Unnamed: 13  Código=  3051011  
0              NaN          NaN      NaN      NaN  
1              NaN          NaN      NaN      NaN  
2              NaN          NaN      NaN      NaN  
3              NaN          NaN      NaN      NaN  
4              NaN          NaN      NaN      NaN  
...            ...          ...      ...      ...  
20600          NaN          NaN      NaN      NaN  
20601          NaN          NaN      NaN      NaN  
20602          NaN          NaN      NaN      NaN  
20603          NaN          NaN      NaN      NaN  
20604          NaN          NaN      NaN      NaN  

[20605 rows x 16 columns]

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20605 entries, 0 to 20604
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Data              20605 non-null  object 
 1   Valor             20160 non-null  float64
 2   Mês               20605 non-null  int64  
 3   Ano               20605 non-null  int64  
 4   Mês-Ano           20605 non-null  object 
 5   ano_valido        19358 non-null  float64
 6   Valor_ano_valido  19292 non-null  float64
 7   Verific_7dias     20605 non-null  int64  
 8   Média 7dias       18742 non-null  float64
 9   Unnamed: 9        0 non-null      float64
 10  Unnamed: 10       0 non-null      float64
 11  Unnamed: 11       0 non-null      float64
 12  Unnamed: 12       0 non-null      float64
 13  Unnamed: 13       0 non-null      float64
 14  Código=           0 non-null      float64
 15  3051011           0 non-null      float64
dtypes: float64(11), int64(3), object(2)
memo

In [7]:
type(df.Data[1])

str

- Need to modify columns names
- Need to convert date into datetime object
- Need to delete unnecessary columns

In [12]:
df[df.Valor.isnull()]

Data  Valor  Mês   Ano Mês-Ano  ano_valido  Valor_ano_valido  \
2342   01/06/67    NaN    6  1967  jun-67      1967.0               NaN   
2343   02/06/67    NaN    6  1967  jun-67      1967.0               NaN   
2344   03/06/67    NaN    6  1967  jun-67      1967.0               NaN   
2345   04/06/67    NaN    6  1967  jun-67      1967.0               NaN   
2346   05/06/67    NaN    6  1967  jun-67      1967.0               NaN   
...         ...    ...  ...   ...     ...         ...               ...   
19718  27/12/14    NaN   12  2014  dez-14      2014.0               NaN   
19719  28/12/14    NaN   12  2014  dez-14      2014.0               NaN   
19720  29/12/14    NaN   12  2014  dez-14      2014.0               NaN   
19721  30/12/14    NaN   12  2014  dez-14      2014.0               NaN   
19722  31/12/14    NaN   12  2014  dez-14      2014.0               NaN   

       Verific_7dias  Média 7dias  Unnamed: 9  Unnamed: 10  Unnamed: 11  \
2342               0          NaN         NaN          NaN          NaN   
2343               0          NaN         NaN          NaN          NaN   
2344               0          NaN         NaN          NaN          NaN   
2345               0          NaN         NaN          NaN          NaN   
2346               0          NaN         NaN          NaN          NaN   
...              ...          ...         ...          ...          ...   
19718              0          NaN         NaN          NaN          NaN   
19719              0          NaN         NaN          NaN          NaN   
19720              0          NaN         NaN          NaN          NaN   
19721              0          NaN         NaN          NaN          NaN   
19722              0          NaN         NaN          NaN          NaN   

       Unnamed: 12  Unnamed: 13  Código=  3051011  
2342           NaN          NaN      NaN      NaN  
2343           NaN          NaN      NaN      NaN  
2344           NaN          NaN      NaN      NaN  
2345           NaN          NaN      NaN      NaN  
2346           NaN          NaN      NaN      NaN  
...            ...          ...      ...      ...  
19718          NaN          NaN      NaN      NaN  
19719          NaN          NaN      NaN      NaN  
19720          NaN          NaN      NaN      NaN  
19721          NaN          NaN      NaN      NaN  
19722          NaN          NaN      NaN      NaN  

[445 rows x 16 columns]

- There are 445 missing values for the precipitation height. Need to delete them or investigate a good method for filling the missing values.

# Cleaning Data

## Define

## Code

## Test